src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/toolbar_js.html.twig

```diff
@@ -9,13 +9,624 @@
+                window.fetch = () => {
+                    const promise = oldFetch.apply(this, arguments);
+                    const url = arguments[0];
+                    const params = arguments[1];
+                    const paramType = Object.prototype.toString.call(arguments[0]);
+
+                    if ('[object Request]' === paramType) {
+                        url = arguments[0].url;
+                        params = {
+                            method: arguments[0].method,
+                            credentials: arguments[0].credentials,
+                            headers: arguments[0].headers,
+                            mode: arguments[0].mode,
+                            redirect: arguments[0].redirect
+                        };
+                    } else {
+                        url = String(url);
+                    }
+
+                    if (!url.match(new RegExp({{ excluded_ajax_paths|json_encode|raw }}))) {
+                        const method = 'GET';
+                        if (params && params.method !== undefined) {
+                            method = params.method;
+                        }
+
+                        const stackElement = {
+                            error: false,
+                            url: url,
+                            method: method,
+                            type: 'fetch',
+                            start: new Date()
+                        };
+
+                        const index = symfonyWebDebugToolbar.#requestStack.push(stackElement) - 1;
+                        promise.then((response) => {
+                            stackElement.duration = new Date() - stackElement.start;
+                            stackElement.error = response.status < 200 || response.status >= 400;
+                            stackElement.statusCode = response.status;
+                            stackElement.profile = response.headers.get('x-debug-token');
+                            stackElement.profilerUrl = response.headers.get('x-debug-token-link');
+                            stackElement.toolbarReplaceFinished = false;
+                            stackElement.toolbarReplace = '1' === response.headers.get('Symfony-Debug-Toolbar-Replace');
+                            symfonyWebDebugToolbar.#finishAjaxRequest(index);
+                        }, (error) => {
+                            stackElement.error = true;
+                            symfonyWebDebugToolbar.#finishAjaxRequest(index);
+                        });
+                        symfonyWebDebugToolbar.#startAjaxRequest(index);
+                    }
+
+                    return promise;
+                };
+            }
+
+            if (window.XMLHttpRequest && XMLHttpRequest.prototype.addEventListener) {
+                const oldXHROpen = window.XMLHttpRequest.prototype.open;
+
+                window.XMLHttpRequest.prototype.open = (method, url, async, user, pass) => {
```

Comment:
the fix for your issue about illegal invocation is to _not_ use an arrow function here, as you want to redefine the method and so have access to the `this` of that method invocation.